src/sonic-dhcpservd/dhcpservd/dhcp_lease.py

```diff
@@ -0,0 +1,147 @@
+        # Store old lease key
+        old_lease_table = self.db_connector.get_state_db_table(DHCP_SERVER_IPV4_LEASE)
+        old_lease_key = set(old_lease_table.keys())
+
+        # 1.1 If start time equal to end time, means lease has been released
+        #     1.1.1 If current lease table has this old lease, delete it
+        #     1.1.2 Else skip
+        # 1.2 Else, means lease valid, save it.
+        for key, value in new_lease.items():
+            if value["lease_start"] == value["lease_end"]:
+                if key in old_lease_key:
+                    self.db_connector.state_db.delete("{}|{}".format(DHCP_SERVER_IPV4_LEASE, key))
+                continue
+            new_key = "{}|{}".format(DHCP_SERVER_IPV4_LEASE, key)
+            for k, v in new_lease[key].items():
+                self.db_connector.state_db.hset(new_key, k, v)
+        # Delete old lease not in new lease set
+        for key in old_lease_key:
+            if key not in new_lease.keys():
+                # Delete entry
+                self.db_connector.state_db.delete("{}|{}".format(DHCP_SERVER_IPV4_LEASE, key))
+        self.last_update_time = datetime.now()
+        self.lock.release()
+
+
+class KeaDhcp4LeaseHandler(LeaseHanlder):
+    def __init__(self, db_connector, lease_file=KEA_LEASE_FILE_PATH, ip_ports_file=DHCP_SERVER_IP_PORTS_FILE):
+        LeaseHanlder.__init__(self, db_connector)
+        self.lease_file = lease_file
+        self.ip_ports_file = ip_ports_file
+
+    def register(self):
+        """
+        Register callback function of signal
+        """
+        signal.signal(signal.SIGUSR1, self._update_lease)
+
+    def _read(self):
+        # Read lease file generated by kea-dhcp4
+        try:
+            with open(self.lease_file, "r", encoding="utf-8") as fb:
+                dq = deque(fb)
+        except FileNotFoundError as err:
+            syslog.syslog(syslog.LOG_ERR, "Cannot find lease file: {}".format(self.lease_file))
+            raise err
+
+        # Read network - dhcp interface map file, this file is generated by dhcp_cfggen while generate config
+        with open(self.ip_ports_file, "r", encoding="utf8")as fp:
+            json_data = json.load(fp)
+            net_intf_map = {}
+            for key, value in json_data.items():
+                net_intf_map[ipaddress.ip_network(key, False)] = value
+            nets = set(net_intf_map.keys())
+
+        new_lease = {}
+        # Get newest lease information of each client
+        while dq:
+            last_row = dq.pop()
+            splits = last_row.split(",")
+            # Skip header
+            if splits[0] == "address":
+                break
+            ip_str = splits[0]
+            mac_address = splits[1]
+            valid_lifetime = splits[3]
+            lease_end = splits[4]
+
+            for net in nets:
+                if ipaddress.ip_address(ip_str) in net:
+                    new_key = "{}|{}".format(net_intf_map[net], mac_address)
```

Comment:
What happens if two vlan configured same subnet?
VLAN1000 - 192.168.0.1/24
VLAN2000 - 192.168.0.1/24

Current idea:
lease <--> Assigned IP --belong to ?---> VLAN?

How about use IP address?
lease <--> client MAC ----FBD----> VLAN?